plutus-tx/src/PlutusTx/Eq.hs

```diff
@@ -80,7 +80,7 @@ instance Eq Data where
     Map _  == _                  = False
     I i == I i'                  = i == i'
     I _ == _                     = False
-    B b == B b'                  = b == b'
+    B b == B b'                  = Builtins.toBuiltin b == Builtins.toBuiltin b'
```

Comment:
This instance should just have been deleted before - people shouldn't be using non-builtin data on-chain, it won't work because there's a non-builtin bytestring in there!